display/ssd1306.h

```diff
@@ -225,15 +229,37 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
 	}
 	if (eof_) {
 	  // STDOUT << "EOF " << frame_count_ << "\n";
-	  screen_ = SCREEN_PLI;
-	  if (frame_count_ == 1) return 5000;
-	  return FillFrameBuffer();
+    if (!SaberBase::IsOn() || static_on_) {
+      screen_ = SCREEN_PLI;
+      if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+      return FillFrameBuffer();
+    }
 	}
 	frame_count_++;
-	if (looped_frames_ > 1 && millis() - loop_start_ > 5000) {
-	  STDOUT << "END OF LOOP\n";
-	  screen_ = SCREEN_PLI;
-	}
+  if (SaberBase::IsOn()) {
+    // Single frame image
+    if (looped_frames_ == 1) {
+      if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+      if (SaberBase::Lockup()) {
+        ShowFile(&IMG_lock);
+        return 3600000;
+      } else {
+        ShowFile(&IMG_on);
+      }
+    // looped image
+    } else {
+      if (looped_frames_ == frame_count_) {
```

Comment:
I'm not sure why we need this line (and the following two)

If we are at the end of an animation, and not in lockup, we start IMG_on.
But wouldn't we already be playing IMG_on? And if it is looped, it would restart automatically.
So why bother?

Reply:
this is how we start IMG_on.  Without these lines, an animated blst or clsh will play forever.

Reply:
That will make the loop play exactly once.
However, if that's what the user wants, they can just make a non-looped animation.
Maybe we should make it so that the play duration can be configured per effect?
That makes a lot of sense for single-frame images as well, you could specify an image to show for "clash", and specify that it should be shown for 200ms or something....

Reply:
Ok, for per-effect durations, would that go in config.ini?

So something like

ProffieOSBootImageDuration
ProffieOSFontImageDuration
ProffieOSClashImageDuration
ProffieOSBlastImageDuration
ProffieOSOnImageDuration

Reply:
Or do we somehow link it to the duration in the blade style for each effect?

Reply:
The blade style durations are going to be all wrong I think. Better to just add config variables for these things. (Unless we could find some way to put them in the file themselves, which would be even better.)

Reply:
added duration variables for all effect types